src/java/org/apache/cassandra/db/compaction/CompactionIterator.java

```diff
@@ -154,6 +154,12 @@ private UnfilteredPartitionIterators.MergeListener listener()
     {
         return new UnfilteredPartitionIterators.MergeListener()
         {
+            @Override
+            public boolean preserveOrder()
+            {
+                return type == OperationType.COMPACTION && controller.cfs.indexManager.hasIndexes();
```

Comment:
Make sure to update this to match the `return null` path in `listener` for the other versions -- perhaps extract this to a method that both sites call?

Reply:
Didn't understand this one, would you mind expanding on this one?

Reply:
@blambov would be great to have more feedback on this

Reply:
Apologies, the comment is not clear at all. The code for choosing to return null is different in C* 5, and at the moment there's no way to see that the code for `preserveOrder` should be updated to match it (the patch will merge cleanly but won't be correct).

We should extract this `type == OperationType.COMPACTION && controller.cfs.indexManager.hasIndexes()` decision into a method, and call it from here as well as `listener()`. That way we will get a merge conflict when this is being ported up.

Reply:
Okay I have updated this.